mypy/checkstrformat.py

```diff
@@ -432,6 +436,37 @@ def check_specs_in_format_call(
                 self.check_placeholder_type(a_type, expected_type, call)
                 self.perform_special_format_checks(spec, call, repl, a_type, expected_type)
 
+    def helpful_check(self, actual_type: ProperType, context: Context) -> bool:
+        fail = False
+        if isinstance(actual_type, Instance):
+            for base in actual_type.type.mro[:-1]:
+                if base.module_name == "builtins":
+                    return True
```

Comment:
does this mean all builtins are assumed to have helpful strings?

Reply:
yeah

Reply:
i think theres plenty that don't. a common example would be accidentally not calling a function
```py
def foo() -> str:
    ...

f"{foo}" # function doesn't have a helpful `__str__` method. but no error because `function` is a builtin type?

Reply:
well, functions are banned, so

Reply:
Actually, come to think of it, stuff like `map` really suck.

Reply:
okay, I blacklisted all the `builtin`s that suck, I think it's pfgl now